test/conditional-pino.test.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const { test } = require('node:test')
+
+test("pino is not require'd if logger is not passed", t => {
+  t.plan(1)
+
+  const fastify = require('..')
+
+  fastify()
+
+  t.assert.ok(!require.cache[require.resolve('pino')])
+})
+
+test("pino is require'd if logger is passed", t => {
+  t.plan(1)
+
+  const fastify = require('..')
+
+  fastify({
+    logger: true
+  })
+
+  t.assert.ok(require.cache[require.resolve('pino')])
+})
+
+test("pino is require'd if loggerInstance is passed", t => {
+  t.plan(1)
+
+  const fastify = require('..')
+
+  const loggerInstance = {
+    fatal: (msg) => { },
+    error: (msg) => { },
+    warn: (msg) => { },
+    info: (msg) => { },
+    debug: (msg) => { },
+    trace: (msg) => { },
+    child: () => loggerInstance
+  }
+
+  fastify({
+    loggerInstance
+  })
+
+  t.assert.ok(require.cache[require.resolve('pino')], true)
```

Comment:
Slight miss, here. The `assert.ok` method's second parameter is a message parameter. I would use `assert.equal` here.

Reply:
@jsumners Updated. Should be fine now.